Content.Server/DeltaV/Implants/Radio/RadioImplantSystem.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Content.Server.Chat.Systems;
+using Content.Server.Radio;
+using Content.Server.Radio.Components;
+using Content.Server.Radio.EntitySystems;
+using Content.Shared.DeltaV.Implants.Radio;
+using Content.Shared.Radio.Components;
+using Robust.Shared.Containers;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+
+namespace Content.Server.DeltaV.Implants.Radio;
+
+/// <inheritdoc />
+public sealed class RadioImplantSystem : SharedRadioImplantSystem
+{
+    [Dependency] private readonly INetManager _netManager = default!;
+    [Dependency] private readonly RadioSystem _radioSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadioImplantComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<RadioImplantComponent, EntInsertedIntoContainerMessage>(OnInsertEncryptionKey);
+        SubscribeLocalEvent<RadioImplantComponent, EntRemovedFromContainerMessage>(OnRemoveEncryptionKey);
+        SubscribeLocalEvent<RadioImplantComponent, RadioReceiveEvent>(OnRadioReceive);
+        SubscribeLocalEvent<HasRadioImplantComponent, EntitySpokeEvent>(OnSpeak);
+    }
+
+    /// <summary>
+    /// Ensures implants with fixed channels work.
+    /// </summary>
+    private void OnMapInit(EntityUid uid, RadioImplantComponent component, MapInitEvent args)
+    {
+        UpdateRadioReception(uid, component);
+    }
+
+    /// <summary>
+    /// Handles the implantee's speech being forwarded onto the radio channel of the implant.
+    /// </summary>
+    private void OnSpeak(EntityUid uid, HasRadioImplantComponent hasRadioImplantComponent, EntitySpokeEvent args)
+    {
+        // not a radio message, or already handled by another radio
+        if (args.Channel is null)
+            return;
+
+        // does the implant have access to the channel the implantee is trying to speak on?
+        if (hasRadioImplantComponent.Implant is { Valid: true }
+            && TryComp<RadioImplantComponent>(hasRadioImplantComponent.Implant, out var radioImplantComponent)
+            && radioImplantComponent.Channels.Contains(args.Channel.ID))
+        {
+            _radioSystem.SendRadioMessage(uid, args.Message, args.Channel.ID, hasRadioImplantComponent.Implant.Value);
+            // prevent other radios they might be wearing from sending the message again
+            args.Channel = null;
+        }
+    }
+
+    /// <summary>
+    /// Handles receiving radio messages and forwarding them to the implantee.
+    /// </summary>
+    private void OnRadioReceive(EntityUid uid, RadioImplantComponent component, ref RadioReceiveEvent args)
+    {
+        if (TryComp(component.Implantee, out ActorComponent? actorComponent))
```

Comment:
get an EntityQuery for this in initialize since this is run a lot
then _actorQuery.TryComp(ent.Comp.Implantee, out var actor))